dhcpv6/dhcpv6.go

```diff
@@ -204,13 +203,9 @@ func IsUsingUEFI(msg *Message) bool {
 	//               7    EFI BC
 	//               8    EFI Xscale
 	//               9    EFI x86-64
-	if opt := msg.GetOneOption(OptionClientArchType); opt != nil {
-		optat := opt.(*OptClientArchType)
-		for _, at := range optat.ArchTypes {
-			// TODO investigate if other types are appropriate
-			if at == iana.EFI_BC || at == iana.EFI_X86_64 {
-				return true
-			}
+	if archTypes := msg.Options.ArchTypes(); archTypes != nil {
```

Comment:
This is changing the current behaviour of the function. If a packet has multiple arch type options it will check only one. I have seen clients sending packets with more than one option type and different info in each of them. I think that in v4 we merge them in the map.

Reply:
No, it doesn't. Previously, this used `GetOneOption`, which only returns one option. The option itself contains multiple arch types in it, though, and that behavior is preserved with the for loop here.

Reply:
`iana.Archs` is a `[]iana.Arch` with a `Contains` function that has the for loop.

Reply:
You are right, so my concern is with `OROs` then.